lib/prbot/github.ex

```diff
@@ -16,7 +16,43 @@ defmodule Prbot.Github do
   end
 
   def getReviews(prNumber) do
-    Tentacat.Pulls.Reviews.list "PartCycleTech", "partcycle-backend", prNumber, client
+    Tentacat.Pulls.Reviews.list "PartCycleTech", "prbot", prNumber, client
+  end
+
+  def getReviewRequests(prNumber) do
+    Tentacat.Pulls.ReviewRequests.list "PartCycleTech", "partcycle-backend", prNumber, client
+  end
+
+  def getRequestedReviewers(prNumber) do
+    for request <- getReviewRequests(prNumber), do: request["login"]
+  end
+
+  def getReviewsByUser(prNumber, username) do
+    for review <- getReviews(prNumber), username == review["user"]["login"], do: review
+  end
+
+  def getLatestReviews(prNumber) do
+    reviewsOldestFirst = Prbot.Github.getReviews(prNumber)
+    reviewsNewestFirst = Enum.reverse(reviewsOldestFirst)
+    Enum.uniq_by(reviewsNewestFirst, fn review -> review["user"]["login"] end)
+  end
+
+  def getStatusOfLatestReviews(prNumber) do
+    # Each status is one of: "APPROVED", "COMMENTED", "CHANGES_REQUESTED"
+    Enum.map(getLatestReviews(prNumber), fn(review) -> review["state"] end)
+  end
+
+  def getStatusOfPR(prNumber) do
+    cond do
+      Enum.all?(getStatusOfLatestReviews(prNumber), fn(status) -> status == "APPROVED" end) ->
+        "APPROVED"
+      Enum.any?(getStatusOfLatestReviews(prNumber), fn(status) -> status == "CHANGES_REQUESTED" end) ->
+        "CHANGES_REQUESTED"
+      Enum.any?(getStatusOfLatestReviews(prNumber), fn(status) -> status == "COMMENTED" end) ->
```

Comment:
Yes. But also, I think we want to implement that "significant events" thing too.